demos/multi_guess_client.py

```diff
@@ -52,7 +52,7 @@ async def main():
         # until they are no longer interested.
 
         # What I want:
-        async with client.get_resource_stream('demo', None) as resource_stream:
+        async with client.stream_resources('demo', None) as resource_stream:
```

Comment:
Now that we have the no-with-needed design, we can get rid of this "What I want" section and correct the comments below.